### PR TITLE
chore: Add BDD test for retrieving documents by versionID

### LINKED ISSUE #1148: BDD test for versionId
Update sidetree-core, add BDD test for retrieving documents by versionId.

### PR DESCRIPTION
**Pull request by sandrask**
Update sidetree-core, add BDD test for retrieving documents by versionId.

Closes #1148

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: test/bdd/features/did-sidetree.feature [hunk 1]
```diff
@@ -227,6 +227,67 @@ Feature:
...
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/operations" to add public key with ID "thirdKey" to DID document
+        Then check for request success
+        Then we wait 2 seconds
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did
+        Then check success response contains "thirdKey"
+
+        # start version queries
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did and version index 0
```
Comment (bstasyszyn) [inline test/bdd/features/did-sidetree.feature:267]: What does version index mean?

Comment (sandrask) [reply test/bdd/features/did-sidetree.feature:267]: just a way of collecting version ids in BDD test - I will change it as per our conversation

Comment (sandrask) [reply test/bdd/features/did-sidetree.feature]: Done

File: test/bdd/did_orb_steps.go [hunk 2]
```diff
@@ -807,6 +821,22 @@ func (d *DIDOrbSteps) resolveDIDDocumentWithCanonicalDID(url string) error {
 	return d.resolveDIDDocumentWithID(url, d.canonicalDID)
 }
 
+func (d *DIDOrbSteps) resolveDIDDocumentWithCanonicalDIDAndVersion(url, version string) error {
+	logger.Infof("resolving did document with canonical did %s and version %s", d.canonicalDID, version)
```
Comment (bstasyszyn) [inline test/bdd/did_orb_steps.go:825]: You can do this:

```
if err := d.state.resolveVarsInExpression(&version); err != nil {
  return err
}

didWithParam := d.canonicalDID + "?versionId=" + versionID


```

File: test/bdd/features/did-sidetree.feature [hunk 3]
```diff
@@ -227,6 +227,71 @@ Feature:
...
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/operations" to create DID document
+        Then check success response contains "#interimDID"
+        Then we wait 2 seconds
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with interim did
+        Then check success response contains "canonicalId"
+        Then the JSON path 'didDocumentMetadata.versionId' of the response is saved to variable "v0"
+
+        When client sends request to "https://orb.domain1.com/sidetree/v1/identifiers" to resolve DID document with canonical did and version "v0"
```
Comment (bstasyszyn) [inline test/bdd/features/did-sidetree.feature:243]: Do this instead:

`. . . canonical did and version "${v0}"
`

**Review by bstasyszyn — APPROVED** [review]

**Review by fqutishat — APPROVED** [review]

**Pull request merged**